src/stress_analyzer.py

```diff
@@ -0,0 +1,157 @@
+        _, stdout1, _ = ssh_client.exec_command(docker_container_id)
+        container_ids = stdout1.read().splitlines()
+        _, stdout2, _ = ssh_client.exec_command(docker_container_cmd)
+        container_commands = stdout2.read().splitlines()
+        for i in range(len(container_ids)):
+            if services == '*' or (container_commands[i] in services):
+                container_id_dict[container_commands[i]] = (vm_ip, container_ids[i])
+
+    return container_id_dict
+
+
+# Returns a tuple of dictionaries of the services and respective container_ids
+# The first dictionary holds the containers to be Stressed
+def get_container_ids_binary(vm_ips, services):
+    container_id_dict = get_container_ids_all(vm_ips, services)
+    stress_dict1 = dict(container_id_dict.items()[len(container_id_dict)/2:])
+    stress_dict2 = dict(container_id_dict.items()[:len(container_id_dict)/2])
+
+    return stress_dict1,stress_dict2
+
+
+### Functions below are for updating container dictionaries
+
+# Returns an updated dictionary of the containers to stress based on type
+# OLD_CONTAINERS is the previous dictionary of containers, and RESULTS are the
+#     results of the stress tests
+def get_updated_container_ids(old_containers, results, stress_type):
+    if stress_type == 'ALL':
+        return # Will Implement later
+    elif stress_type == 'BINARY':
+        # In this case, old_containers should be a tuple of dictionaries
+        return get_updated_container_ids_binary(old_containers, results)
+    return
+
+
+# Old_containers MUST be a tuple of dictionaries (2)
+def get_updated_container_ids_binary(old_containers, results):
+    old_stress_dict1, old_stress_dict2 = old_containers
+    results1, results2 = results
+
+    # Check if iteration ends
+    len1 = len(old_stress_dict1)
+    len2 = len(old_stress_dict2)
+    if (len1 == 1 and len2 == 0) or (len1 == 0 and len2 == 1):
+        return None
+
+    if compare_performance_binary(results1, results2): # Results indicate stress
+        stress_dict1 = dict(old_stress_dict1.items()[len(old_stress_dict1)/2:])
+        stress_dict2 = dict(old_stress_dict1.items()[:len(old_stress_dict1)/2])
+    else:
+        stress_dict1 = dict(old_stress_dict2.items()[len(old_stress_dict2)/2:])
+        stress_dict2 = dict(old_stress_dict2.items()[:len(old_stress_dict2)/2])
+
+    return stress_dict1,stress_dict2
+
+
+# Returns true if the first set of results perform worse than the second
+# Results1,2 must be a tuple (3)!
+# This method should only be used for stress search type BINARY
+# NOTE: Only considering the results from increment 50 for experimental purposes
+def compare_performance_binary(results1, results2):
+    results_cpu1, results_disk1, results_net1 = results1
+    results_cpu2, results_disk2, results_net2 = results2
+    results_list = [results_cpu1, results_disk1, results_net1, results_cpu2, results_disk2, results_net2]
```

Comment:
Results_list would then be a list of 2 tuples, but I wanted to separate the tuples so that I can access the results from different resources easier.